Merge feature/specs

sebGilR opened the pull request:
Specifications for the project have been completed with RSpect in the branch feature/specs.

Commit:
RSpec set up for the project and spec files created

Commit:
player specs complete

carlosprietobarron committed:
Create a new class to control game flow

carlosprietobarron committed:
Creates test examples for game object

Commit:
add specs for board creation and board structure

Commit:
add validation specs for exists, busy, and valid_move methods

Commit:
win and update_cell methods complete

Commit:
fix formatting and styling errors based on rubocop

Commit:
Update README file with RSpec instructions for testing

Commit:
Merge branch 'feature/readme_game_instructions' into feature/specs

Commit:
Fix rubocop error

Commit:
Fix rubocop new added cops warning

sebGilR commented:
Rubocop failure is due to it displaying a warning about new cops added.

mosesogwo reviewed (changes requested):
**Hi @sebGilR, @carlosprietobarron :clap:. Good work with your tests:+1:. Now check the following:**
- The test for your win? method in Board class passes even when I modified the win_combos arrays. This shouldn't be so. You should also test for wins along the diagonals and verticals.
- Besides the new cops that were added, you have modified your rubocop configuration. You should use the original config except where you have been specifically instructed to make changes.


_**Kindly make these changes and submit another code review request.**_

**Any question? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

Commit:
Fix rubocop linting suggestions and add tests

create test for win? diagonally and vertically
verify win? is affected by changes in win_combos array

sebGilR commented:
Thank you for the feedback provided.

- When testing by altering the win_combos array, and removing a win combo used in the tests, the test returns false as expected, the method works fine in these scenarios.
- Tests were created for the diagonal and vertical win possibilities.
- rubocop.yml original file is in use now.

euqueme reviewed (changes requested):
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @sebGilR and @carlosprietobarron :balloon: :lollipop: 

### Thank you for doing the changes requested by the previous TSE, :medal_sports: :medal_sports: I see you've done a great job with this project, even creating a new class to be able to add more test cases so claps to both of you for that :clap: :clap: :clap: please address to the following changes:

#### General

- [x] your linters are showing errors because some cops were added but aren't configured, please configure them and only submit for a code review when you see a green checkmark :heavy_check_mark: 

```
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file:
 - Lint/RaiseException (0.81)
 - Lint/StructNewOverride (0.81)
 - Style/HashEachMethods (0.80)
 - Style/HashTransformKeys (0.80)
 - Style/HashTransformValues (0.80)
```
#### Project specific
[OPTIONAL]
* [x] specs cover positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)) 

- Your project is only covering positive test scenarios and it should cover both positive and negative I'm sorry this wasn't pointed out by the previous reviewer, that's why I'm setting it as optional but I strongly recommend to add it because it was in the project requirements and It'll make your project look complete.

Hint: this is one of your test cases that can be turned into a negative test case:

```ruby
# Positive
    it 'returns false if the passed cell doesn\'t exist in the board' do
      cell = 10
      expect(Board.exists?(cell)).to eql(false)
    end
# Negative
    it "doesn't return true if the passed cell doesn't exist in the board" do
      cell = 10
      expect(Board.exists?(cell)).not_to eql(true)
    end
```
as you can see the description of the it block starts with a negative word and I've used the not_to method.

#### Optional

- [x] your Gamels class should be placed in its own file within the lib directory no classes should be declared in the main.rb file :no_good: this was a requirement in milestone # 3 of the OOP project so I strongly recommend doing this change to respect the OOP rules and so your project stays professional.

### That's it, team! :raised_hands: 

### Please :pray: do the changes :man_technologist: :man_technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

Commit:
Fix rubocop new cops issue

Commit:
fix new cops after rubocop update

Commit:
add missing negative return tests

sebGilR commented:
Requested mandatory changes have been implemented. Optional changes taken into account and negative tests were added were missing.

kobiyoyo reviewed (approved):
# Project Approved :thumbsup:
## Hello Team,  You have done a great job with this project. :clap:
